bin/diagrams

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env node
+
+const canihaz = require('canihaz');
+const path = require('path');
+const fs = require('fs').promises;
+
+const source = path.join(__dirname, '..', 'assets', 'diagrams');
+const template = `
+<!doctype html>
+<html>
+  <head>
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.8.1/css/all.min.css">
+    <style>
+      html, * {
+        font-family: 'monaco';
+      }
+
+      .node {
+        text-align: center;
+      }
+      
+      i.fa {
+        display: block;
+        margin: 5px;
+      }
+
+      .edgeLabel:not(:empty) {
+        padding: 0px 8px;
+      }
+    </style>
+  </head>
+  <body>
+    <div id="container" class="mermaid"></div>
+    <script src="https://unpkg.com/mermaid/dist/mermaid.min.js" charset="utf-8"></script>
+  </body>
+</html>
+`;
+
+async function snapshot(page, file) {
+  try {
+    const chart = await fs.readFile(path.join(source, file), 'utf-8');
+    const output = path.join(source, '..', `${ path.basename(file, '.mmd') }.png`);
+
+    console.log(`Rendering ${ file }`);
+
+    await page.setContent(template, { waitUntil: ['load', 'networkidle0', 'domcontentloaded'] });
+    await page.setViewport({ width: 1920, height: 1080 });
+    await page.evaluate(chart => {
+      window.mermaid.mermaidAPI.initialize({ theme: 'forest' });
+      window.mermaid.mermaidAPI.render('id1', chart, svg => {
+        document.getElementById('container').innerHTML = svg;
+      });
+    }, chart);
+
+    const clip = await page.$eval('#container svg', svg => {
+      const box = svg.getBoundingClientRect();
+      return { x: box.left, y: box.top, width: box.width, height: box.height };
+    });
+
+    // await new Promise(resolve => setTimeout(resolve, 500));
+    await page.screenshot({ path: output, clip, omitBackground: true });
```

Comment:
if mermaid will make svgs, should we just save the svg itself, or do we need a png?

Reply:
hmm that is a good question.

Reply:
Actually if we want icon support and other fancy styling we will need to export to png. Otherwise we will probably be battling Github limitations. Perhaps we can circle back to this if we don't like it. We'll have to generate the diagrams anyway.